app/src/main/java/org/coepi/android/ui/symptoms/SymptomsViewModel.kt

```diff
@@ -10,22 +10,35 @@ import io.reactivex.rxkotlin.withLatestFrom
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
 import io.reactivex.subjects.PublishSubject.create
+import org.coepi.android.cen.CENApi
+import org.coepi.android.cen.CenRepo
+import org.coepi.android.cen.RealmCenDao
+import org.coepi.android.cen.RealmCenKeyDao
+import org.coepi.android.cen.RealmCenReport
+import org.coepi.android.cen.RealmCenReportDao
+import org.coepi.android.cen.SymptomReport
 import org.coepi.android.domain.model.Symptom
 import org.coepi.android.extensions.toLiveData
 import org.coepi.android.extensions.toObservable
 import org.coepi.android.extensions.toUnitObservable
 import org.coepi.android.extensions.toggle
+import org.coepi.android.repo.RealmProvider
 import org.coepi.android.repo.SymptomRepo
 import org.coepi.android.system.log.log
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.Date
 
-class SymptomsViewModel(private val symptomRepo: SymptomRepo) : ViewModel() {
+class SymptomsViewModel(private val symptomRepo: SymptomRepo, private val realmprovider: RealmProvider) : ViewModel() {
```

Comment:
`RealmProvider` is only to be injected in the DAOs, so they can retrieve a `Realm` instance. It shouldn't be used outside of the DAOs.

What you have to inject here is repositories. The repositories in turn use the DAOs to communicate with Realm.